### PR TITLE
Clean up websocket logs

### PR DESCRIPTION
**Pull request by Sriep**
- [x] I added unit tests for any code that added
- [ ] I updated the CHANGELOG.md 
- [x] All IP is original and not copied from another source
- [x] I assign all copyright to Loom Network for the code in the pull request

Remove unnecessary logging of errors when closing WebSockets.
Add mutex to mock query server to remove `-race` errors.
https://github.com/loomnetwork/loomchain/issues/1372

### CODE REVIEW

File: rpc/client.go [hunk 1]
```diff
@@ -53,18 +53,28 @@ type Client struct {
 // reads from this goroutine.
 func (c *Client) readPump(funcMap map[string]eth.RPCFunc, logger log.TMLogger) {
 	defer func() {
+		if r := recover(); r != nil {
+			logger.Error("WebSocket read panicked", "err", r)
+		}
 		c.hub.unregister <- c
-		_ = c.conn.Close()
+		if err := c.conn.Close(); err != nil {
+			logger.Info("Error closing WebSocket read pump", "err", err)
```
Comment (enlight) [inline rpc/client.go]: Why is this `Info` instead of `Error`?

File: rpc/client.go [hunk 2]
```diff
@@ -114,15 +127,19 @@ func (c *Client) writePump(logger log.TMLogger) {
 			}
 
 			if err := c.conn.WriteMessage(websocket.TextMessage, message); err != nil {
-				logger.Error("error writing message to websocket", "err", err)
+				if err.Error() != "io: read/write on closed pipe" {
```
Comment (enlight) [inline rpc/client.go]: Err, nope, not gonna do that. Why does it matter what kind of error it is? Just log it regardless.

File: rpc/client.go [hunk 3]
```diff
@@ -114,15 +127,19 @@ func (c *Client) writePump(logger log.TMLogger) {
...
+				}
 				return
 			}
 
 			n := len(c.send)
 			for i := 0; i < n; i++ {
 				msg := <-c.send
 				if err := c.conn.WriteMessage(websocket.TextMessage, msg); err != nil {
-					logger.Error("error writing message to websocket", "err", err)
+					if err.Error() != "io: read/write on closed pipe" {
```
Comment (enlight) [inline rpc/client.go]: Same here.

File: rpc/client.go (hunk 2 above)
Comment (Sriep) [reply rpc/client.go]: `io: read/write on closed pipe` error occurs whenever the client closes the WebSocket connection. So it's not really an error so much as being informed that the dialogue is over.
Perhaps give an Info message, rather than an error, to record when the WebSocket connect closed?

File: rpc/client.go (hunk 3 above)
Comment (Sriep) [reply rpc/client.go]: `io: read/write on closed pipe` error occurs whenever the client closes the WebSocket connection. So it's not really an error so much as being informed that the dialogue is over.
Perhaps give an Info message, rather than an error, to record when the WebSocket connect closed?

File: rpc/client.go (hunk 2 above)
Comment (enlight) [reply rpc/client.go]: You can check for `err != io.ErrClosedPipe` then. It doesn't seem like logging any of these socket errors as `logger.Error` is particularly useful, maybe `logger.Debug` instead... in case anyone ever needs to debug something. Keep using `logger.Error` for the panics though, those shouldn't be happening often I presume.

Comment (Sriep) [reply rpc/client.go]: Two types of error.
1)  Errors that occur repeatedly during normal operation. (e.g. closed socket and EOF)
2)  Errors that normally never happen.
So report 1) as debug logs and 2) as error logs.

File: rpc/client.go [hunk 4]
```diff
@@ -53,18 +54,30 @@ type Client struct {
...
-			websocket.IsUnexpectedCloseError(err, websocket.CloseGoingAway, websocket.CloseAbnormalClosure)
-			break
+			if websocket.IsUnexpectedCloseError(
+				err,
+				websocket.CloseGoingAway,
+				websocket.CloseAbnormalClosure,
+			) {
+				logger.Error("WebSocket unexpected close error", "err", err)
+			} else {
+				logger.Debug("WebSocket close request", "message", err)
```
Comment (enlight) [inline rpc/client.go]: Which one are you trying to log here, `message` or `err`?

File: rpc/client.go [hunk 5]
```diff
@@ -94,41 +107,52 @@ func (c *Client) writePump(logger log.TMLogger) {
...
 			if err := c.conn.SetWriteDeadline(time.Now().Add(writeWait)); err != nil {
 				logger.Error("error setting write deadline", "err", err)
 			}
 
 			if err := c.conn.WriteMessage(websocket.TextMessage, message); err != nil {
-				logger.Error("error writing message to websocket", "err", err)
+				if err != io.ErrClosedPipe {
+					logger.Error("error writing message to WebSocket", "err", err)
+				} else {
+					logger.Debug("WebSocket connection closed", "message", err)
```
Comment (enlight) [inline rpc/client.go]: Ditto

File: rpc/client.go [hunk 6]
```diff
@@ -94,41 +107,52 @@ func (c *Client) writePump(logger log.TMLogger) {
...
 
 			n := len(c.send)
 			for i := 0; i < n; i++ {
 				msg := <-c.send
 				if err := c.conn.WriteMessage(websocket.TextMessage, msg); err != nil {
-					logger.Error("error writing message to websocket", "err", err)
+					if err != io.ErrClosedPipe {
+						logger.Error("error writing message to WebSocket", "err", err)
+					} else {
+						logger.Debug("WebSocket connection closed", "message", err)
```
Comment (enlight) [inline rpc/client.go]: Ditto

File: rpc/client.go [hunk 7]
```diff
@@ -53,18 +54,30 @@ type Client struct {
...
+			if websocket.IsUnexpectedCloseError(
+				err,
+				websocket.CloseGoingAway,
+				websocket.CloseAbnormalClosure,
+			) {
+				logger.Error("WebSocket unexpected close error", "err", err)
+			} else {
+				logger.Debug("WebSocket close request", "err", err)
+			}
+			return
```
Comment (enlight) [inline rpc/client.go]: There was a `break` here before, now there's a `return`, was this change intentional?

Comment (Sriep) [reply rpc/client.go]: No reason that I remember, both should do the same thing. I guess `return` is more explicit.
